plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -236,34 +242,43 @@ public function add_fields_data() {
 	 */
 	public function register_checkout_field( $options ) {
 		if ( empty( $options['id'] ) ) {
-			return new \WP_Error( 'woocommerce_blocks_checkout_field_id_required', __( 'The field id is required.', 'woo-gutenberg-products-block' ) );
+			return new \WP_Error( 'woocommerce_blocks_checkout_field_id_required', __( 'The field id is required.', 'woocommerce' ) );
 		}
 
 		list( $namespace, $name ) = explode( '/', $options['id'] );
 
 		// Having $name empty means they didn't pass a namespace.
 		if ( empty( $name ) ) {
-			return new \WP_Error( 'woocommerce_blocks_checkout_field_namespace_required', __( 'An id must consist of namespace/name.', 'woo-gutenberg-products-block' ) );
+			return new \WP_Error( 'woocommerce_blocks_checkout_field_namespace_required', __( 'An id must consist of namespace/name.', 'woocommerce' ) );
 		}
 
 		if ( empty( $options['label'] ) ) {
-			return new \WP_Error( 'woocommerce_blocks_checkout_field_label_required', __( 'The field label is required.', 'woo-gutenberg-products-block' ) );
+			return new \WP_Error( 'woocommerce_blocks_checkout_field_label_required', __( 'The field label is required.', 'woocommerce' ) );
 		}
 
 		if ( empty( $options['location'] ) ) {
-			return new \WP_Error( 'woocommerce_blocks_checkout_field_location_required', __( 'The field location is required.', 'woo-gutenberg-products-block' ) );
+			return new \WP_Error( 'woocommerce_blocks_checkout_field_location_required', __( 'The field location is required.', 'woocommerce' ) );
 		}
 
 		if ( ! in_array( $options['location'], array_keys( $this->fields_locations ), true ) ) {
-			return new \WP_Error( 'woocommerce_blocks_checkout_field_location_invalid', __( 'The field location is invalid.', 'woo-gutenberg-products-block' ) );
+			return new \WP_Error( 'woocommerce_blocks_checkout_field_location_invalid', __( 'The field location is invalid.', 'woocommerce' ) );
+		}
+
+		$type = 'text';
+		if ( ! empty( $options['type'] ) ) {
+			if ( ! in_array( $options['type'], $this->supported_field_types, true ) ) {
+				// translators: %1$s is the registered field type, %2$s is a list of supported field types (comma separated).
+				return new \WP_Error( 'woocommerce_blocks_checkout_field_type_unsupported', sprintf( __( 'Registering a field with type "%1$s" is not supported. The supported types are: %2$s.', 'woocommerce' ), esc_html( $options['type'] ), implode( ', ', $this->supported_field_types ) ) );
```

Comment:
What are your thoughts on adding some line breaks for readability-reasons?

```suggestion
				return new \WP_Error(
					'woocommerce_blocks_checkout_field_type_unsupported',
					sprintf(
						__(
							'Registering a field with type "%1$s" is not supported. The supported types are: %2$s.',
							'woocommerce'
						),
						esc_html( $options['type'] ),
						implode( ', ', $this->supported_field_types )
					)
				);
```